src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+### fluid.popLogging()
+
+Pops one `LogLevel` value that was pushed onto the logging level stack by a previous call to `fluid.setLogging`. This will expose the previous element on top of the stack and restore the logging level that had previously been in effect.
+If the stack has only its one original element, this call is a no-op. 
+
+### fluid.passLogLevel(testLogLevel)
+
+Accepts one of the members of the `fluid.logLevel` structure or a `LogLevel` object in general. Returns `true` if
+a message supplied at that log priority would be logged at the current logging level. Clients who
+issue particularly expensive log payload arguments are recommended to guard their logging statements with this
+
+* `testLogLevel {LogLevel}` The log level against which the system's current logging level is to be checked.
+* Returns: `{Boolean}` - `true` if a message supplied at that log priority would be logged at the current logging level.
+
+### fluid.doLog([arg1, ... argn])
+
+The actual implementation to which `fluid.log` messages and other logging generated by the system. Implemented to forward to `console.debug` and `console.log` but may be monkey-patched by 
+clients who require an alternative implementation.
+
+### fluid.fail([arg1, ... argn])
+
+The framework and clients can call this method in order to signal a hard assertion failure. The current contract of `fluid.fail` is that an implementation/log error has been discovered - the
+typical behaviour of the system is to halt or at the very least propagate an unhandled exception to the top process level which is likely to make the system's behaviour unstable. If this failure
+resulted from bad user or external input, do not call this function but instead signal it in a milder, more implementation-specific way - for example by sending an HTTP status code or showing
+a user message. As Infusion evolves to handle a more finely graded variety of dynamic authoring scenarios, this function will be broken up into variants which can signal failures of different severities.
+
+The current system behaviour is to throw a particular subclass of the standard `Error` type, of type `fluid.FluidError`. The behaviour of `fluid.fail` can be customised by contributing handlers to
+the standard event `fluid.failureEvent`
+
+### fluid.failureEvent
+
+Every call to `fluid.fail` triggers a firing of this global event. This is a standard [Infusion Event](InfusionEventSystem.md) with several handlers installed on startup to take on various functions.
+Extra handlers are registered, and the builtin handlers overriden, to customise error handling behaviour in various contexts - for example, when Infusion is running in node.js, is running a 
+[jqUnit test](https://wiki.fluidproject.org/display/docs/Writing+JavaScript+Unit+Tests) ***TODO - obsolete docs for jqUnit*** or is running in [Kettle](https://github.com/fluid-project/kettle) etc.
```

Comment:
Should this TODO be here? perhaps put it in brackets, or something? It's odd, the way it is.